mantidproject/mantid
# Fix beam stop arm masking for certain angles in the ISIS SANS Gui

Fixes #18711 

The underlying issue is a precision issue. The code itself was correct but the definition of a very large cylinder which is being compared with very small values was unfortunate. The large value is intended to simulate a semi-infinite cylinder. The fix defines the length of the cylinder to 10m instead of 1e6m which is still sufficiently large in comparison with all available detector geometries.

**To test:**

Please find the sample data here: smb://olympic/babylon5/Scratch/Anton/LOQ_BUGS
Make sure that the data is in the Mantid path

1. Open the ISIS SANS GUI
2. Set the instrument to LOQ
3. Load the instrument user file: USER_LOQ_164D_M4_Lu_12mm_Changer_MAIN_maskline.txt
4. As a scatter sample set:
5. Press `Load data`
6. Go to the `Masking` tab and press `Display mask`
    * Confirm that the there is a line mask on the detector bank (light blue line)
7. Edit the user file: Change MASK/LINE 8 149.9 to MASK/LINE 8 150.1
8. Reload the user file
9. Close the existing instrument view and repeat step 6 

For release notes see [here](https://github.com/mantidproject/mantid/commit/39f5548a6daf9d502edc16681da2f7bf030d6a68)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
